com.unity.netcode.gameobjects/Runtime/Messaging/NetworkMessageManager.cs

```diff
@@ -829,11 +830,17 @@ internal unsafe void ProcessSendQueues()
                     // Skipping the Verify and sneaking the write mark in because we know it's fine.
                     queueItem.Writer.Handle->AllowedWriteMark = sizeof(NetworkBatchHeader);
 #endif
-                    queueItem.BatchHeader.BatchHash = XXHash.Hash64(queueItem.Writer.GetUnsafePtr() + sizeof(NetworkBatchHeader), queueItem.Writer.Length - sizeof(NetworkBatchHeader));
 
-                    queueItem.BatchHeader.BatchSize = queueItem.Writer.Length;
+
+                    var alignedLength = (int)(Math.Ceiling(queueItem.Writer.Length * k_WordAlignBatchCalc) * 8);
+                    queueItem.Writer.TryBeginWrite(alignedLength);
```

Comment:
It adds a few bytes (0-7 bytes) per batch of messages, not per message.

Unfortunately we can't do that #if because every platform has to speak the same protocol. Can't have one platform add extra bytes to a message that the other platform isn't accounting for on the receiving end. Plus, it's actually the receiving end that needs this, so Windows has to add these bytes if sending to Android... and we don't have any information on whether or not the remote end is an android device.

Reply:
Just for context:
The user scenario was an ARMv7 build running on an Android device connecting to a headless 64bit linux server build (I tested with 64bit windows, but the idea is the same).

Reply:
ah true right.
nvm then resolving